pyscitt/pyscitt/key_vault_sign_client.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+import base64
+import hashlib
+
+from azure.identity import DefaultAzureCredential
+from azure.keyvault.certificates import CertificateClient, KeyVaultCertificate
+from azure.keyvault.keys import KeyClient
+from azure.keyvault.keys.crypto import CryptographyClient, SignatureAlgorithm
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives.asymmetric.utils import encode_dss_signature
+from cryptography.x509 import load_pem_x509_certificate
+
+from . import crypto
+
+
+class KeyVaultSignClient:
+    """MemberIdentity implementation that uses Azure Key Vault."""
+
+    def __init__(self, akv_sign_configuration_dict: dict):
+        self._vault_name = akv_sign_configuration_dict["keyVaultName"]
+        self._vault_url = f"https://{self._vault_name}.vault.azure.net"
+        self._identity_certificate_name = akv_sign_configuration_dict["certificateName"]
+        self._identity_certificate_version = akv_sign_configuration_dict[
+            "certificateVersion"
+        ]
+        self.credential = DefaultAzureCredential()
+        cert_client = CertificateClient(
+            vault_url=self._vault_url, credential=self.credential
+        )
+        self.cert = self._decode_certificate(
+            cert_client.get_certificate_version(
+                certificate_name=self._identity_certificate_name,
+                version=self._identity_certificate_version,
+            )
+        )
+        self.key_id = crypto.get_cert_fingerprint(self.cert)
+
+    @staticmethod
+    def _decode_certificate(cert: KeyVaultCertificate) -> str:
```

Comment:
```suggestion
    def _encode_certificate(cert: KeyVaultCertificate) -> str:
```

Kind of pendantic, but IMO this is *encoding* rather than decoding. It's taking the raw cert bytes and making it into PEM format.